docs/conf.py

```diff
@@ -193,48 +194,51 @@
     edit_on_github_branch = "master"
 edit_on_github_source_root = ""
 edit_on_github_doc_root = "docs"
-
 edit_on_github_skip_regex = '_.*|generated/.*'
-
 github_issues_url = 'https://github.com/sunpy/sunpy/issues/'
 
 # -- Options for the Sphinx gallery -------------------------------------------
-
-if on_rtd and os.environ.get('READTHEDOCS_PROJECT').lower() != 'sunpy':
-    def setup(app):
-        app.warn('The gallery build takes too long on RTD, so the '
-                 'gallery will not be built.  You will probably see '
+if has_sphinx_gallery:
+    extensions += ["sphinx_gallery.gen_gallery"]
+    sphinx_gallery_conf = {
+        'backreferences_dir':
+        'generated{}modules'.format(os.sep),  # path to store the module using example template
+        'filename_pattern':
+        '^((?!skip_).)*$',  # execute all examples except those that start with "skip_"
+        'examples_dirs': os.path.join('..', 'examples'),  # path to the examples scripts
+        'gallery_dirs': os.path.join('generated',
+                                    'gallery'),  # path to save gallery generated examples
+        'default_thumb_file': os.path.join('.', 'logo', 'sunpy_icon_128x128.png'),
+        'reference_url': {
+            'sunpy': None,
+            'astropy': 'http://docs.astropy.org/en/stable/',
+            'matplotlib': 'https://matplotlib.org/',
+            'numpy': 'http://docs.scipy.org/doc/numpy/',
+        },
+        'abort_on_example_error': True,
+        'plot_gallery': True
+    }
+
+def rstjinja(app, docname, source):
+    """
+    Render our pages as a jinja template for fancy templating goodness.
+    """
+    # Make sure we're outputting HTML
+    if app.builder.format != 'html':
+        return
+    src = source[0]
+    if "Current status" in src[:20]:
```

Comment:
also @dpshelio what is the purpose of this check? why not check on docname? I am assuming that is the filename in some form?